src/dune_rules/coq_lib.ml

```diff
@@ -173,6 +176,9 @@ module DB = struct
 
     let create_from_stanza =
       let create_from_stanza_impl (coq_db, db, dir, (s : Coq_stanza.Theory.t)) =
+        (* boot libraries cannot depend on other theories *)
+        if s.boot && List.is_non_empty s.buildable.theories then
+          Error.boot_has_deps ();
```

Comment:
If you want to force the error eagerly, why don't you do it at parsing time?

Reply:
Actually I was wondering about this; when would the best moment be for this error to be raised?

Reply:
Validation is something we do as early as possible. So if this is a programmer error, let's give the user an error quickly and with a good location. In this case, the error depends only on local properties of the stanza. So let's do it there.

The reason why we delay resolution errors is to make composed workspaces more usable. For example, you want to vendor some big project but only use one small theory inside it. You don't care that the other theories in the project have missing dependencies. `Resolve.t` is how we make that work.

Reply:
Ok, I'll move it earlier. In fact, I'm gonna tweak the code a bit more and remove that check from this PR.